src/sqlfluff/core/linter/linter.py

```diff
@@ -267,7 +272,25 @@ def parse_noqa(comment: str, line_no: int):
                             )
                     rules: Optional[Tuple[str, ...]]
                     if rule_part != "all":
-                        rules = tuple(r.strip() for r in rule_part.split(","))
+                        # Rules can be globs therefore we compare to the rule_set to expand the globs.
+                        unexpanded_rules = tuple(
+                            r.strip() for r in rule_part.split(",")
+                        )
+                        expanded_rules = []
+                        for r in unexpanded_rules:
+                            if r in {"PRS", "LXR", "TMP"}:
+                                # Don't expand special error types.
+                                if r not in expanded_rules:
+                                    expanded_rules.append(r)
+                            else:
+                                expanded_rules.extend(
+                                    [
+                                        x
+                                        for x in fnmatch.filter(rule_codes, r)
+                                        if x not in expanded_rules
+                                    ]
+                                )
```

Comment:
Bit worried about hardcoding the list of special error types.

Shouldn't we just say if the unexpended rule contains a `*` then expand? If not don't expand?

Reply:
sure, glob matching actually has more patterns than just `*` (e.g. `L01[12]` matches L011 and L012) so we can't just filter on that. 

let me change this section to 
```python
expanded_rule = [
    x
    for x in fnmatch.filter(rule_codes, r)
    if x not in expanded_rules
]
expanded_rules.extend(expanded_rule) if expanded_rule else expanded_rules.append(r)
```

That way we expand if we can else we assume the user is referencing a special error type.